src/Grpc.Net.Client/Balancer/Subchannel.cs

```diff
@@ -273,9 +273,17 @@ private async Task ConnectTransportAsync()
                     }
                 }
 
-                if (await _transport.TryConnectAsync(connectContext).ConfigureAwait(false))
+                switch (await _transport.TryConnectAsync(connectContext).ConfigureAwait(false))
                 {
-                    return;
+                    case ConnectResult.Success:
+                        return;
+                    case ConnectResult.Timeout:
```

Comment:
I had exactly the same issue but with SocketException (for example in case of DNS resolution failure). After the SocketException the channel became unusable because every grpc call fails with the exception.

I would be good to cover this situation as well and in generic way.

Reply:
When that error happens, the connection should continue trying to connect. If your app is configured to write logs at a trace level, you should see `StartingConnectBackoff` logs.

Reply:
Unfortunately, I don't have trace level enabled for gprc in prod. What I see is that my Grpc client became unusable for ~6 hours. Interesting that for the same application instance all other operations (CosmosDb calls, some non-grpc related HTTP calls and so on) were working fine.

From [here](https://learn.microsoft.com/en-us/aspnet/core/grpc/loadbalancing?view=aspnetcore-7.0#configure-custom-resolvers-and-load-balancers):
> By default, a DNS resolver is refreshed if a connection is interrupted. 

May be the DNS refresh wasn't triggered after the first SocketException?

Reply:
I can't say without seeing logging. There are a lot of asynchronous actions happening inside the client, especially when connections drop and the resolver updates. That is why there is detailed logging of everything that is happening.

Are you using an up-to-date version of Grpc.Net.Client? Try 2.50.0. If the problem persists then I can take a look at the logs and try to determine what is going on.